src/core/shape_inference/include/detection_output_shape_inference.hpp

```diff
@@ -0,0 +1,343 @@
+                                  num_loc_classes * 4,
+                                  "). Current value is: ",
+                                  box_logits_pshape_2nd_dim,
+                                  ".");
+            num_prior_boxes = box_logits_pshape_2nd_dim / (num_loc_classes * 4);
+        }
+    }
+    if (class_preds_pshape.rank().is_static()) {
+        NODE_VALIDATION_CHECK(op,
+                              class_preds_pshape.size() == 2,
+                              "Class predictions rank must be 2. Got ",
+                              class_preds_pshape.size());
+        if ((!dim_num_images_updated || dim_num_images.is_dynamic()) && class_preds_pshape[0].is_static()) {
+            dim_num_images = class_preds_pshape[0].get_length();
+            dim_num_images_updated = true;
+        } else {
+            NODE_VALIDATION_CHECK(
+                op,
+                class_preds_pshape[0].compatible(dim_num_images),
+                "Class predictions' first dimension is not compatible with batch size.  Current value is: ",
+                class_preds_pshape[0],
+                ", expected: ",
+                dim_num_images,
+                ".");
+        }
+        if (class_preds_pshape[1].is_static() && num_classes) {
+            auto class_preds_pshape_2nd_dim = class_preds_pshape[1].get_length();
+            if (!num_prior_boxes) {
+                NODE_VALIDATION_CHECK(op,
+                                      class_preds_pshape_2nd_dim % num_classes == 0,
+                                      "Class predictions' second dimension must be a multiply of num_classes (",
+                                      num_classes,
+                                      "). Current value is: ",
+                                      class_preds_pshape_2nd_dim,
+                                      ".");
+                num_prior_boxes = class_preds_pshape_2nd_dim / num_classes;
+            } else {
+                NODE_VALIDATION_CHECK(op,
+                                      class_preds_pshape_2nd_dim == num_prior_boxes * num_classes,
+                                      "Class predictions' second dimension must be equal to num_prior_boxes * "
+                                      "num_classes (",
+                                      num_prior_boxes * num_classes,
+                                      "). Current value is: ",
+                                      class_preds_pshape_2nd_dim,
+                                      ".");
+            }
+        }
+    }
+    if (proposals_pshape.rank().is_static()) {
+        NODE_VALIDATION_CHECK(op,
+                              proposals_pshape.size() == 3,
+                              "Proposals rank must be 3. Got ",
+                              proposals_pshape.size());
+        NODE_VALIDATION_CHECK(op,
+                              proposals_pshape[0].compatible(1) || proposals_pshape[0].compatible(dim_num_images),
```

Comment:
nit: it's a `broadcast_merge` of dimension

Reply:
Did you mean the broadcast_merge( result ,  proposals_pshape[0], dim_num_images)?  I think this can't be the same with the above.   Assuming proposals_pshape[0] = [1, 12], dim_num_images = [13, 20]. The original check can pass but the 
 broadcast_merge( result ,  proposals_pshape[0], dim_num_images) would fail.